pdp/ast/jast/selector.go

```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/infobloxopen/themis/pdp"
 )
 
-func (ctx context) unmarshalSelector(d *json.Decoder) (pdp.LocalSelector, error) {
+func (ctx context) unmarshalSelector(d *json.Decoder) (pdp.Expression, error) {
```

Comment:
The same as for YAML parser - length of PIP url should not be generally limited and it's better make map string to selector parser instead of "if" statement.

Reply:
`length of PIP url should not be generally limited`

Did you mean this:
```
                loc := strings.Split(id.Opaque, "/")
                if len(loc) != 2 {
```
If that's what you mean...
The concept is that the url specifies `service_name/query_type`, and parameters passed in as attributes. I am not sure there's a need to allow a more general url.

Reply:
Basically, we name this field URI and it brings following format: `scheme:[//[user[:password]@]host[:port]][/path][?query][#fragment]`. The first thing which matters to PDP is `scheme`. As soon as `scheme` is "local" meaning of rest is defined by PDP as PDP handles whole story. For "pip" scheme PDP should only care of `[user[:password]@]host[:port]]` to reach PIP server. Anything else it should pass to PIP server to not narrow possible implementations of latest.